compiler/qsc_linter/src/lints/ast.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use super::push_lint;
+use crate::linter::ast::declare_ast_lints;
+use qsc_ast::ast::{BinOp, ExprKind, Lit, StmtKind};
+use qsc_data_structures::span::Span;
+
+declare_ast_lints! {
+    (DivisionByZero, LintLevel::Allow, "attempt to divide by zero"),
+    (NeedlessParens, LintLevel::Warning, "unnecessary parentheses"),
+    (RedundantSemicolons, LintLevel::Warning, "redundant semicolons"),
```

Comment:
I think it would be helpful to include miette-style diagnostic help like we do for other errors. Lints in particular benefit from some additional explanatory text for the user. For example, "redundant semicolons" could come with the additional help text, "remove the redundant semicolons for clarity."

Reply:
Yes, you are right. I just added help messages to the lints